smart-contracts/contracts/mixins/MixinLockManagerRole.sol

```diff
@@ -0,0 +1,47 @@
+pragma solidity ^0.5.0;
+
+import '@openzeppelin/upgrades/contracts/Initializable.sol';
+import '@openzeppelin/contracts-ethereum-package/contracts/access/Roles.sol';
+
+
+contract MixinLockManagerRole {
+  using Roles for Roles.Role;
+
+  event LockManagerAdded(address indexed account);
+  event LockManagerRemoved(address indexed account);
+
+  Roles.Role private _LockManagers;
+
+  function _initializeMixinLockManagerRole(address sender) internal {
+    if (!isLockManager(sender)) {
+      _addLockManager(sender);
+    }
+  }
+
+  modifier onlyLockManager() {
+    require(isLockManager(msg.sender), 'MixinLockManager: caller does not have the LockManager role');
+    _;
+  }
+
+  function isLockManager(address account) public view returns (bool) {
+    return _LockManagers.has(account);
+  }
+
+  function addLockManager(address account) public onlyLockManager {
+    _addLockManager(account);
```

Comment:
This pattern is overkill.  The purpose is to expose _addLockManager to other contracts.  That's something that makes sense for a library implementation but not really for a custom / modified version.

Reply:
I've cleaned this up to remove this pattern. Note that as a result I've had to now access `lockManagers` directly in the `_initializeMixinLockManagerRole` function, as we can no longer call the internal function `_addLockManager` here.